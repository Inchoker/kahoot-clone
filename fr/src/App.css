.App {
  text-align: center;
}

h1 {
  font-size: 2rem;
}

div {
  margin: 1rem 0;
}

label {
  display: block;
  margin: 0.5rem 0;
}
/* Global styling for better readability */
*,
*::before,
*::after {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

body {
  font-family: Arial, sans-serif;
  background-color: #f9f9f9;
  color: #333;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  margin: 0;
}

/* Container styling */
.add-question {
  background: #fff;
  padding: 2rem;
  width: 100%;
  max-width: 500px;
  border-radius: 8px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
}

/* Heading styling */
.add-question h2 {
  margin-bottom: 1.5rem;
  font-size: 1.8rem;
  color: #444;
  text-align: center;
}

/* Form layout */
form {
  display: grid;
  gap: 1.2rem;
}

/* Label styling */
label {
  font-weight: bold;
  margin-bottom: 0.4rem;
  display: inline-block;
  color: #555;
}

/* Input styling */
input[type="text"] {
  width: 100%;
  padding: 0.8rem;
  border: 1px solid #ddd;
  border-radius: 5px;
  font-size: 1rem;
  transition: border-color 0.3s;
}

input[type="text"]:focus {
  outline: none;
  border-color: #007BFF;
}

/* Options input styling */
.options-container {
  display: grid;
  gap: 0.5rem;
}

/* Button styling */
button[type="submit"] {
  background-color: #007BFF;
  color: white;
  padding: 0.8rem 1.2rem;
  border: none;
  border-radius: 5px;
  cursor: pointer;
  font-size: 1rem;
  transition: background-color 0.3s;
}

button[type="submit"]:hover {
  background-color: #0056b3;
}

/* Message styling */
.message {
  text-align: center;
  font-size: 1rem;
  padding: 0.5rem;
  border-radius: 5px;
  margin-top: 1rem;
}

.message.success {
  background-color: #d4edda;
  color: #155724;
}

.message.error {
  background-color: #f8d7da;
  color: #721c24;
}
/* App.css */
.navbar {
  background-color: #ffffff;
  padding: 1rem;
  display: flex;
  justify-content: center;
  gap: 20px;
}

.nav-link {
  color: #000000;
  text-decoration: none;
  padding: 0.5rem 1rem;
  font-weight: bold;
  transition: background 0.3s ease, color 0.3s ease;
  border-radius: 5px;
}

.nav-link:hover {
  background-color: #000000;
  color: #ffffff;
}

.active-link {
  background-color: #000000;
  color: #ffffff;
}
