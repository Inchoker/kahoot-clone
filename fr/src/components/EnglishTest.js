import React, { useEffect, useState } from 'react';
import io from 'socket.io-client';

const socket = io("http://localhost:5000");

const QuizComponent = () => {
    const [currentQuestion, setCurrentQuestion] = useState(null);
    const [selectedOption, setSelectedOption] = useState('');
    const [score, setScore] = useState(0);

    useEffect(() => {
        socket.on('question', (newQuestion) => {
            console.log(newQuestion)
            setCurrentQuestion(newQuestion);
            setSelectedOption('');
        });

        socket.on('scoreUpdate', (newScore) => {
            setScore(newScore);
        });
        socket.emit('joinQuiz','quiz')

        return () => {
            console.log('here')
            socket.off('question');
            socket.off('scoreUpdate');
            socket.emit('leaveQuiz');
        };
    }, []);

    const handleAnswer = () => {
        if (selectedOption) {
            socket.emit('answer', selectedOption);
            setSelectedOption('');
        }
    };

    return (
        <div style={{ padding: '20px' }}>
            <h1>Real-Time Quiz</h1>
            {currentQuestion ? (
                <div>
                    <h2>{currentQuestion.question}</h2>
                    <p><strong>Category:</strong> {currentQuestion.category}</p>
                    <p><strong>Difficulty:</strong> {currentQuestion.difficulty}</p>
                    <div>
                        {currentQuestion.options.map((option, index) => (
                            <div key={index}>
                                <input
                                    type="radio"
                                    id={option}
                                    name="quizOption"
                                    value={option}
                                    checked={selectedOption === option}
                                    onChange={(e) => setSelectedOption(e.target.value)}
                                />
                                <label htmlFor={option}>{option}</label>
                            </div>
                        ))}
                    </div>
                    <button onClick={handleAnswer}>Submit Answer</button>
                </div>
            ) : (
                <p>Waiting for questions...</p>
            )}
            <h3>Your Score: {score}</h3>
        </div>
    );
};

export default QuizComponent;
