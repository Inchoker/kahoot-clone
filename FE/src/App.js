import React, {createContext, useState} from 'react';
import './App.css';
import EnglishTest from './Components/EnglishTest';
import AddQuestion from "./Components/AddQuestions";
import { BrowserRouter as Router, Routes, Route, NavLink } from "react-router-dom";
import {QueryClient, QueryClientProvider} from "react-query";

const PlayerContext = createContext();
function App() {
    const queryClient = new QueryClient();
    const [player,setPlayer] = useState(null);
  return (
      <div className="App">
          <QueryClientProvider client={queryClient}>
              <PlayerContext.Provider value={{player,setPlayer}}>
                  <Router>
                      <nav className="navbar">
                          <NavLink to="/" className="nav-link" activeClassName="active-link" exact>
                              Home
                          </NavLink>
                          <NavLink to="/add" className="nav-link" activeClassName="active-link">
                              Add more questions
                          </NavLink>
                      </nav>

                      <Routes>
                          <Route path="/" element={<EnglishTest />} />
                          <Route path="/add" element={<AddQuestion />} />
                      </Routes>
                  </Router>
              </PlayerContext.Provider>
          </QueryClientProvider>
      </div>
);
}

export default App;
export {PlayerContext};
