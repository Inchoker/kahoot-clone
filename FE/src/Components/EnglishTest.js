import React, {useContext, useEffect, useState} from 'react';
import socket from "../SocketIoClient/SocketIo";
import {PlayerContext} from "../App";

const QuizComponent = () => {
    const [currentQuestion, setCurrentQuestion] = useState(null);
    const [selectedOption, setSelectedOption] = useState('');
    const [score, setScore] = useState([]);
    const {player, setPlayer} = useContext(PlayerContext)

    useEffect(() => {
        socket.on('question', (newQuestion) => {
            console.log(newQuestion)
            setCurrentQuestion(newQuestion);
            setSelectedOption('');
        });
        socket.on('connect',()=>{
            setPlayer(socket.id.slice(0,4));
            socket.emit('joinQuiz','quiz')
        })
        socket.on('disconnect',()=>{
            setPlayer('left. Refresh to play');
        })

        socket.on('leaderboardUpdate', (newScore) => {
            setScore(newScore);
        });


        return () => {

            socket.off('question');
            socket.off('scoreUpdate');
            socket.emit('leaveQuiz');
        };
    }, [setPlayer]);

    const handleAnswer = () => {
        if (selectedOption) {
            socket.emit('answer', selectedOption,player);
            setSelectedOption('');
        }
    };


    return (
        <div style={{ padding: '20px' }}>
            <h1>Real-Time Quiz. You are player {player}</h1>
            {currentQuestion ? (
                <div>
                    <h2>{currentQuestion.question}</h2>
                    <p><strong>Category:</strong> {currentQuestion.category}</p>
                    <p><strong>Difficulty:</strong> {currentQuestion.difficulty}</p>
                    <div>
                        {currentQuestion.options.map((option, index) => (
                            <div key={index}>
                                <input
                                    type="radio"
                                    id={option}
                                    name="quizOption"
                                    value={option}
                                    checked={selectedOption === option}
                                    onChange={(e) => setSelectedOption(e.target.value)}
                                />
                                <label htmlFor={option}>{option}</label>
                            </div>
                        ))}
                    </div>
                    <button onClick={handleAnswer}>Submit Answer</button>
                </div>
            ) : (
                <p>Waiting for questions...</p>
            )}
            {score.map((item, index) => <h3 id={index}>{item.value}: {item.score}</h3>)}

        </div>
    );
};

export default QuizComponent;
