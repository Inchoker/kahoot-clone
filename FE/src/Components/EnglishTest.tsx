import React, { useContext, useEffect, useState } from 'react';
import socket from '../SocketIoClient/SocketIo';
import { PlayerContext } from '../App';

interface Question {
  question: string;
  options: string[];
  correctAnswer: string;
  category: string;
  difficulty: string;
}

interface ScoreItem {
  value: string;
  score: number;
}

interface PlayerContextType {
  player: string;
  setPlayer: React.Dispatch<React.SetStateAction<string>>;
}

const QuizComponent: React.FC = () => {
  const [currentQuestion, setCurrentQuestion] = useState<Question | null>(null);
  const [selectedOption, setSelectedOption] = useState<string>('');
  const [score, setScore] = useState<ScoreItem[]>([]);
  const [nameInput, setNameInput] = useState('');
  const { player, setPlayer } = useContext(PlayerContext) as PlayerContextType;

  useEffect(() => {
    socket.on('question', (newQuestion: Question) => {
      setCurrentQuestion(newQuestion);
      setSelectedOption('');
    });

    socket.on('connect', () => {
      if (player) {
        socket.emit('joinQuiz', 'quiz');
      }
    });

    socket.on('disconnect', () => {
      setPlayer('left. Refresh to play');
    });

    socket.on('leaderboardUpdate', (newScore: ScoreItem[]) => {
      setScore(newScore);
    });

    return () => {
      socket.off('question');
      socket.off('leaderboardUpdate');
      socket.emit('leaveQuiz');
    };
  }, [player, setPlayer]);

  const handleAnswer = () => {
    if (selectedOption) {
      socket.emit('answer', selectedOption, player);
      setSelectedOption('');
      setCurrentQuestion(null)
    }
  };

  const handleNameSubmit = () => {
    if (nameInput.trim()) {
      setPlayer(nameInput.trim());
      socket.emit('joinQuiz', 'quiz');
    }
  };

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      handleNameSubmit()
    }
  };

  if (!player || player === 'left. Refresh to play') {
    return (
      <div className="flex flex-col items-center justify-center min-h-screen bg-gray-100 p-4">
        <h1 className="text-2xl font-bold mb-4">Enter Your Name to Join</h1>
        <input
          type="text"
          className="border border-gray-300 rounded px-4 py-2 mb-3 w-64"
          value={nameInput}
          onChange={(e) => setNameInput(e.target.value)}
          placeholder="Your name"
          onKeyDown={handleKeyDown}
        />
        <button
          onClick={handleNameSubmit}
          className="bg-blue-500 text-white px-6 py-2 rounded hover:bg-blue-600"
        >
          Join
        </button>
      </div>
    );
  }

  return (
    <div className="min-h-screen bg-white text-gray-800 p-6">
      <h1 className="text-3xl font-bold mb-4">
        Real-Time Quiz <span className="text-blue-600">({player})</span>
      </h1>

      {currentQuestion ? (
        <div className="mb-6 border p-4 rounded shadow-md bg-gray-50">
          <h2 className="text-xl font-semibold mb-2">{currentQuestion.question}</h2>
          <p className="text-sm mb-1"><strong>Category:</strong> {currentQuestion.category}</p>
          <p className="text-sm mb-4"><strong>Difficulty:</strong> {currentQuestion.difficulty}</p>

          <div className="space-y-2 mb-4">
            {currentQuestion.options.map((option, index) => (
              <label
                key={index}
                className={`block border p-2 rounded cursor-pointer ${
                  selectedOption === option ? 'bg-blue-100 border-blue-500' : 'hover:bg-gray-100'
                }`}
              >
                <input
                  type="radio"
                  name="quizOption"
                  value={option}
                  className="mr-2"
                  checked={selectedOption === index+1+""}
                  onChange={(e) => setSelectedOption(index+1+"")}
                />
                {option}
              </label>
            ))}
          </div>

          <button
            onClick={handleAnswer}
            className="bg-green-500 text-white px-6 py-2 rounded hover:bg-green-600"
          >
            Submit Answer
          </button>
        </div>
      ) : (
        <p className="text-lg font-medium">Waiting for questions...</p>
      )}

      <div className="mt-6">
        <h2 className="text-xl font-bold mb-2">Leaderboard</h2>
        <ul className="space-y-1">
          {score.map((item, index) => (
            <li
              key={index}
              className="flex justify-between bg-gray-100 p-2 rounded shadow-sm"
            >
              <span className="font-medium">{item.value}</span>
              <span>{item.score} pts</span>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default QuizComponent;
