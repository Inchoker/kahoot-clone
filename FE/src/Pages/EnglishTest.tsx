import React, { useContext, useEffect, useState } from 'react';
import socket from '../SocketIoClient/SocketIo';
import { PlayerContext } from '../App';

interface Question {
  question: string;
  options: string[];
  correctAnswer: string;
  category: string;
  difficulty: string;
}

interface ScoreItem {
  value: string;
  score: number;
}

interface PlayerContextType {
  player: string;
  setPlayer: React.Dispatch<React.SetStateAction<string>>;
}

const QuizComponent: React.FC = () => {
  const [currentQuestion, setCurrentQuestion] = useState<Question | null>(null);
  const [selectedOption, setSelectedOption] = useState<string>('');
  const [score, setScore] = useState<ScoreItem[]>([]);
  const [nameInput, setNameInput] = useState('');
  const { player, setPlayer } = useContext(PlayerContext) as PlayerContextType;

  // Socket setup
  useEffect(() => {
    socket.on('question', (newQuestion: Question) => {
      setCurrentQuestion(newQuestion);
      setSelectedOption('');
    });

    socket.on('connect', () => {
      if (player) socket.emit('joinQuiz', 'quiz');
    });

    socket.on('disconnect', () => {
      setPlayer('left. Refresh to play');
    });

    socket.on('leaderboardUpdate', (newScore: ScoreItem[]) => {
      setScore(newScore.sort((a,b)=>b.score-a.score));
    });

    return () => {
      socket.off('question');
      socket.off('leaderboardUpdate');
      socket.emit('leaveQuiz');
    };
  }, [player, setPlayer]);

  const handleAnswer = () => {
    if (selectedOption) {
      socket.emit('answer', selectedOption, player);
      setSelectedOption('');
      setCurrentQuestion(null);
    }
  };

  const handleNameSubmit = () => {
    if (nameInput.trim()) {
      setPlayer(nameInput.trim());
      socket.emit('joinQuiz', 'quiz');
    }
  };

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter') {
      e.preventDefault();
      handleNameSubmit();
    }
  };

  if (!player || player === 'left. Refresh to play') {
    return (
      <div className="flex flex-col items-center justify-center min-h-screen bg-gradient-to-br from-blue-50 to-blue-100 p-4">
        <h1 className="text-3xl font-bold mb-4 text-gray-800">Join the Quiz</h1>
        <input
          type="text"
          className="border border-gray-300 rounded-md px-4 py-2 mb-3 shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-400 !w-72"
          value={nameInput}
          onChange={(e) => setNameInput(e.target.value)}
          placeholder="Enter your name"
          onKeyDown={handleKeyDown}
        />
        <button
          onClick={handleNameSubmit}
          className="bg-blue-600 text-white px-6 py-2 rounded-md shadow hover:bg-blue-700 transition"
        >
          Join
        </button>
      </div>
    );
  }

  return (
    <div className="bg-gray-50 min-h-screen flex flex-col xl:flex-row xl:min-h-screen xl:w-full xl:overflow-hidden max-w-[1280px] mx-auto">
      
      {/* Left Panel - Question */}
      <div className="w-full xl:w-2/3 p-6 xl:p-12 flex flex-col justify-center">
        <h1 className="text-3xl font-bold mb-6 text-center xl:text-left text-gray-800">
          Quiz Time <span className="text-blue-600">({player})</span>
        </h1>

        {currentQuestion ? (
          <div className="bg-white p-6 rounded-2xl shadow-md border border-gray-200">
            <h2 className="text-2xl font-semibold mb-4 text-gray-700">{currentQuestion.question}</h2>
            <div className="text-sm text-gray-500 mb-4 flex gap-4">
              <p><strong>Category:</strong> {currentQuestion.category}</p>
              <p><strong>Difficulty:</strong> {currentQuestion.difficulty}</p>
            </div>

            <div className="grid grid-cols-1 sm:grid-cols-2 gap-4 mb-6">
              {currentQuestion.options.map((option, index) => {
                const value = (index + 1).toString();
                return (
                  <label
                    key={index}
                    className={`border p-3 rounded-lg text-gray-800 font-medium cursor-pointer transition ${
                      selectedOption === value
                        ? 'bg-blue-100 border-blue-500'
                        : 'hover:bg-gray-100'
                    }`}
                  >
                    <input
                      type="radio"
                      name="quizOption"
                      value={value}
                      className="mr-2"
                      checked={selectedOption === value}
                      onChange={(e) => setSelectedOption(e.target.value)}
                    />
                    {option}
                  </label>
                );
              })}
            </div>

            <button
              onClick={handleAnswer}
              className="w-full bg-green-600 text-white py-3 rounded-md font-semibold hover:bg-green-700 transition"
            >
              Submit Answer
            </button>
          </div>
        ) : (
          <div className="text-center text-lg font-medium text-gray-600">
            Waiting for questions...
          </div>
        )}
      </div>

      {/* Right Panel - Leaderboard */}
      <div className="w-full xl:w-1/3 bg-white p-6 xl:p-12 border-t xl:border-t-0 xl:border-l border-gray-200 flex flex-col justify-between min-h-[320px]">
        <h2 className="text-xl font-bold mb-4 text-gray-800 text-center xl:text-left">Leaderboard</h2>
        <ul className="space-y-3 overflow-auto flex-grow">
          {score.map((item, index) => (
            <li
              key={index}
              className="flex items-center justify-between px-4 py-2 bg-gray-100 rounded-md shadow-sm"
            >
              <div className="flex items-center gap-2">
                <span className="text-sm font-bold text-blue-600">#{index + 1}</span>
                <span className="text-gray-800">{item.value}</span>
              </div>
              <span className="text-gray-600">{item.score} pts</span>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default QuizComponent;
