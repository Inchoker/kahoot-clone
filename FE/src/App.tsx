import React, { createContext, useState, Dispatch, SetStateAction } from 'react';
import './App.css';
import EnglishTest from './Pages/EnglishTest';
import AddQuestions from "./Pages/AddQuestions";
import { BrowserRouter as Router, Routes, Route, NavLink } from "react-router-dom";
import { QueryClient, QueryClientProvider } from "react-query";

interface PlayerContextType {
  player: string | null;
  setPlayer: Dispatch<SetStateAction<string | null>>;
}

const PlayerContext = createContext<PlayerContextType | undefined>(undefined);

function App() {
  const queryClient = new QueryClient();
  const [player, setPlayer] = useState<string | null>(null);

  return (
    <div className="App">
      <QueryClientProvider client={queryClient}>
        <PlayerContext.Provider value={{ player, setPlayer }}>
          <Router>
            <nav className="navbar">
              <NavLink to="/" className="nav-link" end>
                Home
              </NavLink>
              <NavLink to="/add" className="nav-link">
                Add more questions
              </NavLink>
            </nav>
            <Routes>
              <Route path="/" element={<EnglishTest />} />
              <Route path="/add" element={<AddQuestions />} />
            </Routes>
          </Router>
        </PlayerContext.Provider>
      </QueryClientProvider>
    </div>
  );
}

export default App;
export { PlayerContext, PlayerContextType };
