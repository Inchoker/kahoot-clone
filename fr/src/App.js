import React from 'react';
import './App.css';
import EnglishTest from './components/EnglishTest';
import AddQuestion from "./components/AddQuestions";
import { BrowserRouter as Router, Routes, Route, NavLink } from "react-router-dom";

function App() {
  return (
      <div className="App">
          <Router>
              <nav className="navbar">
                  <NavLink to="/" className="nav-link" activeClassName="active-link" exact>
                      Home
                  </NavLink>
                  <NavLink to="/add" className="nav-link" activeClassName="active-link">
                      Add more questions
                  </NavLink>
              </nav>

              <Routes>
                  <Route path="/" element={<EnglishTest />} />
                  <Route path="/add" element={<AddQuestion />} />
              </Routes>
          </Router>
      </div>
);
}

export default App;
