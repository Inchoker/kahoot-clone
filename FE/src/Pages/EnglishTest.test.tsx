import React from 'react';
import { render } from '@testing-library/react';
import EnglishTest from './EnglishTest';
import { PlayerContext } from '../App';

jest.mock('../SocketIoClient/SocketIo', () => ({
  on: jest.fn(),
  emit: jest.fn(),
  off: jest.fn()
}));

describe('QuizComponent snapshot', () => {
  it('matches snapshot when user is not joined', () => {
    const mockSetPlayer = jest.fn();

    const { container } = render(
      <PlayerContext.Provider value={{ player: '', setPlayer: mockSetPlayer }}>
        <EnglishTest />
      </PlayerContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot when user has joined', () => {
    const mockSetPlayer = jest.fn();

    const { container } = render(
      <PlayerContext.Provider value={{ player: 'Alice', setPlayer: mockSetPlayer }}>
        <EnglishTest />
      </PlayerContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
