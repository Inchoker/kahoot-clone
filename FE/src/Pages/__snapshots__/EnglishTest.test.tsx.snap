// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizComponent snapshot matches snapshot when user has joined 1`] = `
<div>
  <div
    class="bg-gray-50 min-h-screen flex flex-col xl:flex-row xl:min-h-screen xl:w-full xl:overflow-hidden max-w-[1280px] mx-auto"
  >
    <div
      class="w-full xl:w-2/3 p-6 xl:p-12 flex flex-col justify-center"
    >
      <h1
        class="text-3xl font-bold mb-6 text-center xl:text-left text-gray-800"
      >
        Quiz Time 
        <span
          class="text-blue-600"
        >
          (
          Alice
          )
        </span>
      </h1>
      <div
        class="text-center text-lg font-medium text-gray-600"
      >
        Waiting for questions...
      </div>
    </div>
    <div
      class="w-full xl:w-1/3 bg-white p-6 xl:p-12 border-t xl:border-t-0 xl:border-l border-gray-200 flex flex-col justify-between min-h-[320px]"
    >
      <h2
        class="text-xl font-bold mb-4 text-gray-800 text-center xl:text-left"
      >
        Leaderboard
      </h2>
      <ul
        class="space-y-3 overflow-auto flex-grow"
      />
    </div>
  </div>
</div>
`;

exports[`QuizComponent snapshot matches snapshot when user is not joined 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center min-h-screen bg-gradient-to-br from-blue-50 to-blue-100 p-4"
  >
    <h1
      class="text-3xl font-bold mb-4 text-gray-800"
    >
      Join the Quiz
    </h1>
    <input
      class="border border-gray-300 rounded-md px-4 py-2 mb-3 shadow-sm focus:outline-none focus:ring-2 focus:ring-blue-400 !w-72"
      placeholder="Enter your name"
      type="text"
      value=""
    />
    <button
      class="bg-blue-600 text-white px-6 py-2 rounded-md shadow hover:bg-blue-700 transition"
    >
      Join
    </button>
  </div>
</div>
`;
